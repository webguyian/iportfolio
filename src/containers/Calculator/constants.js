export const add = '+';
export const allClear = 'AC';
export const backspace = 'Backspace';
export const clear = 'C';
export const decimal = '.';
export const divide = '/';
export const divideAlt = '\u00f7';
export const enter = 'Enter';
export const equals = '=';
export const multiply = '*';
export const multiplyAlt = '\u00d7';
export const percent = '%';
export const positiveNegative = '+/-';
export const subtract = '-';
export const zero = '0';

export const controls = {
  add,
  allClear,
  clear,
  decimal,
  divide,
  divideAlt,
  equals,
  multiply,
  multiplyAlt,
  percent,
  positiveNegative,
  subtract
};

export const allOperators = [
  add,
  allClear,
  clear,
  decimal,
  divide,
  divideAlt,
  equals,
  multiply,
  multiplyAlt,
  percent,
  positiveNegative,
  subtract
];

export const operators = [
  add,
  allClear,
  clear,
  divide,
  divideAlt,
  multiply,
  multiplyAlt,
  positiveNegative,
  subtract
];

export const controlLabels = {
  [allClear]: 'All clear',
  [clear]: 'Clear',
  [positiveNegative]: 'Positive/negative',
  [percent]: 'Percent',
  [multiplyAlt]: 'Multiply',
  [divideAlt]: 'Divide',
  [add]: 'Add',
  [subtract]: 'Subtract',
  [equals]: 'Equals',
  [decimal]: 'Decimal'
};

export const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];

export const keys = {
  backspace,
  enter
};

export const operatorKeys = [backspace, enter]
  .concat(...allOperators)
  .concat(numbers);
