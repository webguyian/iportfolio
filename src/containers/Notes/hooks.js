import { useEffect, useRef, useState } from 'react';
import { useSwipeable } from 'react-swipeable';

export const useRefFocus = (condition = true, initial = null) => {
  const ref = useRef(initial);

  useEffect(() => {
    const element = ref && ref.current;

    if (condition && element) {
      // Add focus to element
      element.focus();
    }
  }, [condition, ref]);

  return ref;
};

export const useSwipeOffset = (initial = 0) => {
  const [offset, setOffset] = useState(initial);
  const maxOffset = -30;

  const handleSwipe = eventInfo => {
    if (eventInfo.dir !== 'Down') {
      // Exit early if not down swipe
      return false;
    }

    setOffset(eventInfo.deltaY);
  };

  const handleSwipeUp = () => {
    if (!offset) {
      // Exit early with no offset
      return false;
    }

    setOffset(0);
  };

  const handlers = useSwipeable({
    onSwipedUp: handleSwipeUp,
    onSwiping: handleSwipe,
    trackMouse: true
  });

  return [handlers, offset < maxOffset, setOffset];
};
