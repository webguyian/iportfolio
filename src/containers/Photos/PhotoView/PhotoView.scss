.photo-view {
  @include stretch-column-nowrap;
  position: relative;
  background-color: $black-darker;

  &-top-bar,
  &-bottom-bar {
    @include flex-bar;
    background-color: $black-dark;
  }

  &-top-bar {
    justify-content: space-between;
    padding: $gutter;

    > .ui-link--back,
    > .ui-btn {
      margin-right: 0;
      color: $blue;
      font-size: $font-size-m;
      cursor: pointer;
    }

    > .ui-link--back {
      &:hover,
      &:focus {
        color: lighten($blue, 10%);
      }
    }
  }

  &-bottom-bar {
    z-index: 2;
    position: relative;
    margin-top: auto;

    .ui-icon,
    .svg-inline--fa {
      width: 28px;
      height: 28px;
      color: $blue;
      font-size: 28px;
    }
  }

  &-image-wrapper {
    display: flex;
    justify-content: center;
  }

  &-image {
    pointer-events: none;
    user-select: none;
  }

  .ui-btn {
    &:hover,
    &:focus {
      > .ui-icon,
      > .svg-inline--fa {
        color: lighten($blue, 10%);
      }
    }

    &[disabled] {
      color: $gray-medium;
    }

    &--edit {
      padding: 0 $gutter;
    }
  }

  &-controls {
    position: absolute;
    bottom: 70px;
    width: 100%;
    padding: 0 8px;
    transform: translateY(calc(100% + 70px));
    transition: transform 0.4s;

    &--show {
      transform: translateY(0);
    }

    > .ui-btn--anchor-block {
      transition: background-color 0.2s;
      border-radius: $border-radius-l;
      background-color: $gray-dark;
      color: $blue;

      &:hover,
      &:focus {
        background-color: darken($gray-dark, 2%);
      }

      &:first-child {
        color: $red;
      }

      &:last-child {
        margin-top: 8px;
      }
    }
  }
}
