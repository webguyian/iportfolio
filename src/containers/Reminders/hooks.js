import { useEffect, useState, useRef } from 'react';
import { useSwipeable } from 'react-swipeable';

export const useLocalStorage = (key, initialValue) => {
  const getValue = () => {
    try {
      // Get item from local storage with key
      const item = window.localStorage.getItem(key);

      // Parse stored json or if none return initialValue
      return item ? JSON.parse(item) : initialValue;
    } catch (error) {
      // If error return initialValue
      return initialValue;
    }
  };
  const [storedValue, setStoredValue] = useState(getValue);
  const setValue = value => {
    try {
      // Allow value to be a function so we have same API as useState
      const valueToStore =
        value instanceof Function ? value(storedValue) : value;

      // Save state
      setStoredValue(valueToStore);
      // Save to local storage
      window.localStorage.setItem(key, JSON.stringify(valueToStore));
    } catch (error) {
      // Drop error
    }
  };

  return [storedValue, setValue, getValue];
};

export const useRefFocus = (node = null) => {
  const ref = useRef(node);

  useEffect(() => {
    const element = ref && ref.current;

    if (element) {
      // Add focus to element
      element.focus();
    }
  }, [ref]);

  return ref;
};

export const useRefControlledFocus = (focused, node = null) => {
  const ref = useRef(node);

  useEffect(() => {
    const element = ref && ref.current;

    if (element && focused) {
      element.focus();
    }
  }, [focused]);

  return ref;
};

export const useReminders = (reminders, setReminders) => {
  const remindersRef = useRef(reminders);

  useEffect(() => {
    // Update reminders in ref
    remindersRef.current = reminders;
  });

  useEffect(() => {
    return () => {
      const { current } = remindersRef;
      const updatedReminders = current.filter(reminder =>
        reminder.value.trim()
      );

      // Trigger callback without empty reminders
      setReminders(updatedReminders);
    };
  }, [remindersRef]);

  return remindersRef;
};

export const useSwipeOffset = (callback, initial = 0) => {
  const [offset, setOffset] = useState(initial);

  const handleSwipe = eventInfo => {
    if (eventInfo.dir !== 'Left') {
      // Exit early if not left swipe
      return false;
    }

    setOffset(eventInfo.deltaX);
  };

  const handleSwipeLeft = eventInfo => {
    const { deltaX } = eventInfo;

    if (deltaX > 80) {
      callback(eventInfo);
    }
  };

  const handleSwipeRight = () => {
    if (!offset) {
      // Exit early with no offset
      return false;
    }

    setOffset(0);
  };

  const handlers = useSwipeable({
    onSwipedLeft: handleSwipeLeft,
    onSwipedRight: handleSwipeRight,
    onSwiping: handleSwipe,
    trackMouse: true
  });

  return [handlers, offset, setOffset];
};
