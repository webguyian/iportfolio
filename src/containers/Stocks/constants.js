export const initialStocks = [
  { symbol: 'AAPL', displaySymbol: 'AAPL', description: 'APPLE INC' },
  { symbol: 'AMZN', displaySymbol: 'AMZN', description: 'AMAZON.COM INC' },
  { symbol: 'FB', displaySymbol: 'FB', description: 'FACEBOOK INC-CLASS A' },
  { symbol: 'GOOG', displaySymbol: 'GOOG', description: 'ALPHABET INC-CL C' },
  { symbol: 'MSFT', displaySymbol: 'MSFT', description: 'MICROSOFT CORP' }
];

export const API_CANDLESTICK = 'https://finnhub.io/api/v1/stock/candle';

export const API_NEWS = 'https://finnhub.io/api/v1/news';

export const API_QUOTE = 'https://finnhub.io/api/v1/quote';

export const API_TOKEN = 'bojsah7rh5rcji5m4udg';

export const API_EXCHANGE = `https://finnhub.io/api/v1/stock/symbol?exchange=US&token=${API_TOKEN}`;
