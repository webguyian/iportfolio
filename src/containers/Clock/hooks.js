import { useEffect, useRef, useState } from 'react';
import { useHistory } from 'react-router-dom';

import { initialValues } from './constants';
import { getExpiration, getExpirationValues, getSeconds } from './helpers';

export const useInitialRoute = (match, path) => {
  const history = useHistory();

  useEffect(() => {
    if (match.isExact) {
      history.push(path);
    }
  }, []);

  return history;
};

export const useStorage = (key, currentValue, deleteKey, keepUpdating) => {
  const getValue = () => {
    try {
      // Get item from local storage with key
      const item = window.localStorage.getItem(key);

      // Parse stored JSON or if none return undefined
      return item ? JSON.parse(item) : undefined;
    } catch (error) {
      // If error return undefined
      return undefined;
    }
  };

  const value = getValue();
  const localRef = useRef(value);
  const [storedValue, setStoredValue] = useState(value);

  const setValue = nextValue => {
    try {
      // Save to state and local storage
      setStoredValue(nextValue);

      if (deleteKey && !nextValue[deleteKey]) {
        // Remove item from local storage
        window.localStorage.removeItem(key);
      } else {
        // Update local storage and include timestamp
        nextValue.timestamp = Date.now();
        window.localStorage.setItem(key, JSON.stringify(nextValue));
      }
    } catch (error) {
      // Drop error
    }
  };

  useEffect(() => {
    // Update local value
    localRef.current = currentValue || storedValue;

    if (keepUpdating) {
      // Set update value in local storage
      setValue(localRef.current);
    }
  }, [currentValue, storedValue]);

  useEffect(() => {
    return () => {
      if (key && localRef.current) {
        setValue(localRef.current);
      }
    };
  }, []);

  return localRef.current;
};

export const useLaps = (initialLaps, timer, timerRunning) => {
  const [laps, setLaps] = useState(initialLaps);
  const hasLaps = laps.some(lap => lap > 0);
  const storage = useStorage(
    'stopwatch',
    {
      laps,
      timer,
      timerRunning
    },
    'timer'
  );

  const updateLaps = () => {
    if (timerRunning) {
      const hasEmptyLaps = laps.some(lap => lap === 0);

      if (hasEmptyLaps) {
        setLaps(updatedLaps => [
          0,
          ...updatedLaps.slice(0, updatedLaps.length - 1)
        ]);
      } else {
        setLaps(updatedLaps => [0, ...updatedLaps]);
      }
    } else {
      setLaps(initialLaps);
    }
  };

  useEffect(() => {
    if (storage && storage.laps) {
      setLaps(storage.laps);
    }
  }, []);

  useEffect(() => {
    if (!timerRunning || !timer) {
      return;
    }

    const lapLength = laps.length;

    if (!hasLaps && timer > 10) {
      setLaps(updatedLaps => [timer, ...updatedLaps]);
    }

    if (lapLength > 5) {
      setLaps(updatedLaps => {
        const truncatedLaps = updatedLaps.slice(1, lapLength);
        const offset = truncatedLaps.reduce((a, b) => a + b, 0);

        return [timer - offset, ...truncatedLaps];
      });
    }
  }, [timer, timerRunning, hasLaps]);

  return [laps, updateLaps, hasLaps];
};

export const useStopwatch = (startTime = 0, start = false) => {
  const [timerStarted, startTimer] = useState(start);
  const [interval, setLocalInterval] = useState(null);
  const [offset, setOffset] = useState(0);
  const timer = useRef(startTime);
  const storage = useStorage('stopwatch');
  const toggleTimer = () => startTimer(!timerStarted);
  const resetTimer = () => {
    timer.current = 0;
    return timer.current;
  };
  const getUpdatedTime = cache => {
    const now = Date.now();
    const diff = now - cache.timestamp;

    return cache.timer + diff;
  };

  useEffect(() => {
    if (storage && storage.timer) {
      // Get updated time from local storage
      if (storage.timerRunning) {
        timer.current = getUpdatedTime(storage);
        toggleTimer();
      } else {
        timer.current = storage.timer;
      }
    }
  }, []);

  useEffect(() => {
    let localOffset = offset;

    if (timerStarted && !interval) {
      localOffset = Date.now();

      const getDelta = () => {
        const now = Date.now();
        const delta = now - localOffset;

        localOffset = now;

        setOffset(now);
        return delta;
      };

      const localInterval = window.setInterval(() => {
        timer.current += getDelta();
      }, 1);

      setLocalInterval(localInterval);
    }

    if (!timerStarted && interval) {
      window.clearInterval(interval);
      setLocalInterval(null);
    }
  }, [timerStarted, timer, interval]);

  return [timer.current, timerStarted, toggleTimer, resetTimer];
};

export const useDuration = () => {
  const [values, setValues] = useState(null);
  const storage = useStorage('duration', values, null, true);
  const setDuration = (key, value) => {
    setValues({
      ...values,
      [key]: value
    });
  };
  const cachedValues = storage || values;
  const duration = cachedValues && getSeconds(cachedValues);

  useEffect(() => {
    if (storage) {
      // Update state with values from local storage
      setValues(storage);
    } else {
      // Set default initial values
      setValues(initialValues);
    }
  }, []);

  return [duration, setDuration, cachedValues, setValues];
};

export const useCountdown = (h = 0, m = 0, s = 0) => {
  const [seconds, setSeconds] = useState(s);
  const [minutes, setMinutes] = useState(m);
  const [hours, setHours] = useState(h);
  const allSeconds = getSeconds({ hours, minutes, seconds });

  const subtractHour = () => {
    setHours(prevHours => {
      if (prevHours === 0) {
        return 23;
      } else if (prevHours > 0) {
        return prevHours - 1;
      }
      return 0;
    });
  };

  const subtractMinute = () => {
    setMinutes(prevMinutes => {
      if (prevMinutes === 0) {
        subtractHour();
        return 59;
      } else if (prevMinutes > 0) {
        return prevMinutes - 1;
      }
      return 0;
    });
  };

  const subtractSecond = () => {
    setSeconds(prevSeconds => {
      if (prevSeconds === 0) {
        subtractMinute();
        return 59;
      } else if (prevSeconds > 0) {
        return prevSeconds - 1;
      }
      return 0;
    });
  };

  const setCountdown = values => {
    if (!values) {
      setHours(0);
      setMinutes(0);
      setSeconds(0);
      return;
    }

    setHours(values.hours);
    setMinutes(values.minutes);
    setSeconds(values.seconds);
  };

  return [
    {
      allSeconds,
      hours,
      minutes,
      seconds
    },
    setCountdown,
    subtractSecond
  ];
};

export const useTimer = duration => {
  const [started, setStarted] = useState(false);
  const [running, setRunning] = useState(false);
  const [countdown, setCountdown, startCountdown] = useCountdown();
  const intervalRef = useRef();
  const storage = useStorage(
    'timer',
    {
      ...countdown,
      duration,
      running,
      started
    },
    'allSeconds'
  );

  const pause = () => {
    if (intervalRef.current) {
      setRunning(() => false);
      clearInterval(intervalRef.current);
      intervalRef.current = undefined;
    }
  };

  const reset = () => {
    pause();
    setCountdown(null);
    setStarted(false);
  };

  const calculateExpiration = expiration => {
    const distance = expiration - Date.now();
    const expirationValues = getExpirationValues(distance);

    if (countdown.allSeconds < 0) {
      reset();
    } else {
      setCountdown(expirationValues);
    }

    return expirationValues;
  };

  const start = expiration => {
    if (expiration && !intervalRef.current) {
      calculateExpiration(expiration);
    }
  };

  const resume = () => {
    if (!intervalRef.current) {
      setStarted(() => true);
      setRunning(() => true);
      intervalRef.current = setInterval(startCountdown, 1000);
    }
  };

  const restart = () => {
    const expiration = getExpiration(duration);

    reset();
    start(expiration);
  };

  const toggle = () => {
    if (!intervalRef.current) {
      resume();
    } else {
      pause();
    }
  };

  useEffect(() => {
    const expiration = getExpiration(duration, storage);

    start(expiration);

    if (storage) {
      if (storage.running) {
        resume();
      } else if (storage.started) {
        setStarted(true);
      }

      if (duration !== storage.duration) {
        // Remove timer from local storage
        window.localStorage.removeItem('timer');
      }
    }

    return pause;
  }, [duration]);

  return {
    ...countdown,
    duration,
    start,
    started,
    pause,
    reset,
    resume,
    restart,
    running,
    toggle
  };
};

export const useRadialOffset = (timer, radius) => {
  const circumference = (2 * radius * 22) / 7;
  const { allSeconds, duration } = timer;
  const [initialSeconds] = useState(allSeconds);
  const [offset, setOffset] = useState(0);

  useEffect(() => {
    if (!offset) {
      const initialOffset = (allSeconds / duration) * circumference;
      const updatedOffset = circumference - Math.round(initialOffset);

      if (updatedOffset === circumference || updatedOffset < 0) {
        // Exit early
        return;
      }

      setOffset(updatedOffset);
    }
  }, [allSeconds, duration, offset]);

  return [offset, initialSeconds];
};
