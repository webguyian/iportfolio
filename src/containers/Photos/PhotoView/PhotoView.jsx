import React from 'react';
import { object } from 'prop-types';
import cx from 'classnames';

import Button from 'components/Button/Button';
import Link from 'components/Link/Link';
import Text from 'components/Text/Text';

import { heartOutline } from 'modules/photos/constants';
import {
  formatCreatedDate,
  formatCreatedDateRelative
} from 'modules/photos/helpers';
import { usePhoto, useSwipeOffset } from 'modules/photos/hooks';

const PhotoView = props => {
  const [photo, actions, showControls] = usePhoto(props.photo);
  const [handlers] = useSwipeOffset(actions.onSwipe);
  const baseClass = 'photo-view';
  const controlsClass = `${baseClass}-controls`;
  const showControlsClass = showControls && `${controlsClass}--show`;

  if (!photo) {
    return null;
  }

  const isFavorite = photo.metadata.favorited;

  return (
    <div className={baseClass}>
      <div className={`${baseClass}-top-bar`}>
        <Link to="/photos/gallery" back onClick={actions.onBack} />
        <Text>{formatCreatedDateRelative(photo)}</Text>
        <Button className="ui-btn--edit" disabled>
          Edit
        </Button>
      </div>
      <div className={`${baseClass}-image-wrapper`} {...handlers}>
        <img
          className={`${baseClass}-image`}
          alt={`Photo taken ${formatCreatedDate(photo)}`}
          src={photo.image}
        />
      </div>
      <div className={cx(controlsClass, showControlsClass)}>
        <Button modifier="anchor-block" onClick={actions.onDelete}>
          Delete Photo
        </Button>
        <Button modifier="anchor-block" onClick={actions.onCancel}>
          Cancel
        </Button>
      </div>
      <div className={`${baseClass}-bottom-bar`}>
        <Button icon="envelope" onClick={actions.onShare}>
          Share
        </Button>
        {isFavorite ? (
          <Button icon="heart" onClick={actions.onFavorite}>
            Unfavorite
          </Button>
        ) : (
          <Button onClick={actions.onFavorite}>{heartOutline}</Button>
        )}
        <Button icon="trash-alt" onClick={actions.onConfirmDelete}>
          Delete
        </Button>
      </div>
    </div>
  );
};

PhotoView.propTypes = {
  photo: object.isRequired
};

export default PhotoView;
