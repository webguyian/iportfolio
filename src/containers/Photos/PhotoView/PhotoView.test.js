import React from 'react';
import { render } from '@testing-library/react';

import { mockPhoto } from 'modules/photos/constants';
import * as hooks from 'modules/photos/hooks';

import PhotoView from './PhotoView';

describe('<PhotoView />', () => {
  const props = { photo: mockPhoto };
  const actions = {
    onBack: jest.fn(),
    onCancel: jest.fn(),
    onConfirmDelete: jest.fn(),
    onDelete: jest.fn(),
    onFavorite: jest.fn(),
    onShare: jest.fn()
  };
  const usePhoto = hooks.usePhoto;

  hooks.usePhoto = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
    // Reset to original implementation before each test
    hooks.usePhoto.mockImplementation(usePhoto);
  });

  it('renders correctly', () => {
    const { asFragment } = render(<PhotoView {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with photo', () => {
    hooks.usePhoto.mockReturnValue([mockPhoto, actions, false]);
    const { asFragment } = render(<PhotoView {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with unfavorited photo', () => {
    const photo = {
      ...mockPhoto,
      metadata: {
        ...mockPhoto.metadata,
        favorited: false
      }
    };

    hooks.usePhoto.mockReturnValue([photo, actions, false]);
    const { asFragment } = render(<PhotoView {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with controls', () => {
    hooks.usePhoto.mockReturnValue([mockPhoto, actions, true]);
    const { asFragment } = render(<PhotoView {...props} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
