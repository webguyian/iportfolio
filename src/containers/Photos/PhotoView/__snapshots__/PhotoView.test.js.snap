// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoView /> renders correctly 1`] = `<DocumentFragment />`;

exports[`<PhotoView /> renders correctly with controls 1`] = `
<DocumentFragment>
  <div
    class="photo-view"
  >
    <div
      class="photo-view-top-bar"
    >
      <mock-link
        back="true"
        to="/photos/gallery"
      />
      <span
        class="ui-text ui-text--body"
      >
        0 seconds ago
      </span>
      <button
        class="ui-btn ui-btn--edit"
        disabled=""
        type="button"
      >
        Edit
      </button>
    </div>
    <div
      class="photo-view-image-wrapper"
    >
      <img
        alt="Photo taken Sep 29, 2019 at 12:00 AM"
        class="photo-view-image"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII="
      />
    </div>
    <div
      class="photo-view-controls photo-view-controls--show"
    >
      <button
        class="ui-btn ui-btn--anchor-block"
        type="button"
      >
        Delete Photo
      </button>
      <button
        class="ui-btn ui-btn--anchor-block"
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      class="photo-view-bottom-bar"
    >
      <button
        class="ui-btn ui-btn--with-icon"
        type="button"
      >
        <i
          class="ui-icon ui-icon-envelope fas fa-envelope"
        />
        <span
          class="ui-text ui-text--accessible"
        >
          Share
        </span>
      </button>
      <button
        class="ui-btn ui-btn--with-icon"
        type="button"
      >
        <i
          class="ui-icon ui-icon-heart fas fa-heart"
        />
        <span
          class="ui-text ui-text--accessible"
        >
          Unfavorite
        </span>
      </button>
      <button
        class="ui-btn ui-btn--with-icon"
        type="button"
      >
        <i
          class="ui-icon ui-icon-trash-alt fas fa-trash-alt"
        />
        <span
          class="ui-text ui-text--accessible"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<PhotoView /> renders correctly with photo 1`] = `
<DocumentFragment>
  <div
    class="photo-view"
  >
    <div
      class="photo-view-top-bar"
    >
      <mock-link
        back="true"
        to="/photos/gallery"
      />
      <span
        class="ui-text ui-text--body"
      >
        0 seconds ago
      </span>
      <button
        class="ui-btn ui-btn--edit"
        disabled=""
        type="button"
      >
        Edit
      </button>
    </div>
    <div
      class="photo-view-image-wrapper"
    >
      <img
        alt="Photo taken Sep 29, 2019 at 12:00 AM"
        class="photo-view-image"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII="
      />
    </div>
    <div
      class="photo-view-controls"
    >
      <button
        class="ui-btn ui-btn--anchor-block"
        type="button"
      >
        Delete Photo
      </button>
      <button
        class="ui-btn ui-btn--anchor-block"
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      class="photo-view-bottom-bar"
    >
      <button
        class="ui-btn ui-btn--with-icon"
        type="button"
      >
        <i
          class="ui-icon ui-icon-envelope fas fa-envelope"
        />
        <span
          class="ui-text ui-text--accessible"
        >
          Share
        </span>
      </button>
      <button
        class="ui-btn ui-btn--with-icon"
        type="button"
      >
        <i
          class="ui-icon ui-icon-heart fas fa-heart"
        />
        <span
          class="ui-text ui-text--accessible"
        >
          Unfavorite
        </span>
      </button>
      <button
        class="ui-btn ui-btn--with-icon"
        type="button"
      >
        <i
          class="ui-icon ui-icon-trash-alt fas fa-trash-alt"
        />
        <span
          class="ui-text ui-text--accessible"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<PhotoView /> renders correctly with unfavorited photo 1`] = `
<DocumentFragment>
  <div
    class="photo-view"
  >
    <div
      class="photo-view-top-bar"
    >
      <mock-link
        back="true"
        to="/photos/gallery"
      />
      <span
        class="ui-text ui-text--body"
      >
        0 seconds ago
      </span>
      <button
        class="ui-btn ui-btn--edit"
        disabled=""
        type="button"
      >
        Edit
      </button>
    </div>
    <div
      class="photo-view-image-wrapper"
    >
      <img
        alt="Photo taken Sep 29, 2019 at 12:00 AM"
        class="photo-view-image"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII="
      />
    </div>
    <div
      class="photo-view-controls"
    >
      <button
        class="ui-btn ui-btn--anchor-block"
        type="button"
      >
        Delete Photo
      </button>
      <button
        class="ui-btn ui-btn--anchor-block"
        type="button"
      >
        Cancel
      </button>
    </div>
    <div
      class="photo-view-bottom-bar"
    >
      <button
        class="ui-btn ui-btn--with-icon"
        type="button"
      >
        <i
          class="ui-icon ui-icon-envelope fas fa-envelope"
        />
        <span
          class="ui-text ui-text--accessible"
        >
          Share
        </span>
      </button>
      <button
        class="ui-btn"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart fa-w-16"
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
            fill="currentColor"
          />
        </svg>
      </button>
      <button
        class="ui-btn ui-btn--with-icon"
        type="button"
      >
        <i
          class="ui-icon ui-icon-trash-alt fas fa-trash-alt"
        />
        <span
          class="ui-text ui-text--accessible"
        >
          Delete
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
